core/src/test/kotlin/org/evomaster/core/output/TestCaseWriterTest.kt

```diff
@@ -0,0 +1,289 @@
+        expectedLines.add("}")
+
+        assertEquals(expectedLines.toString(), lines.toString())
+    }
+
+
+    @Test
+    fun testPrimaryKeyColumn() {
+        val idColumn = Column("Id", "INTEGER", 10, primaryKey = true)
+        val nameColumn = Column("Name", "VARCHAR", 10, primaryKey = false)
+
+        val aTable = Table("myTable", setOf(idColumn, nameColumn), HashSet<ForeignKey>())
+
+
+        val id = 0L
+
+        val integerGene = IntegerGene(idColumn.name, 42, 0, 10)
+        val primaryKeyGene = SqlPrimaryKeyGene(idColumn.name, "myTable", integerGene, 10)
+        val stringGene = StringGene(nameColumn.name, "nameValue", 0, 10)
+
+        val insertIntoTableAction = DbAction(aTable, setOf(idColumn, nameColumn), id, listOf(primaryKeyGene, stringGene))
+
+        val (format, baseUrlOfSut, ei) = buildEvaluatedIndividual(mutableListOf(insertIntoTableAction))
+
+        val test = TestCase(test = ei, name = "test")
+
+        val writer = TestCaseWriter()
+
+        val lines = writer.convertToCompilableTestCode(format, test, baseUrlOfSut)
+
+        val expectedLines = Lines()
+        expectedLines.add("@Test")
+        expectedLines.add("public void test() throws Exception {")
+        expectedLines.indent()
+        expectedLines.add("List<InsertionDto> insertions = sql().insertInto(\"myTable\").d(\"Id\", 42).d(\"Name\", \"nameValue\").dtos();")
+        expectedLines.add("controller.execInsertionsIntoDatabase(insertions);")
+        expectedLines.deindent()
+        expectedLines.add("}")
+
+        assertEquals(expectedLines.toString(), lines.toString())
+    }
+
+
+    @Test
+    fun testForeignKeyColumn() {
+        val idColumn = Column("Id", "INTEGER", 10, primaryKey = true)
+        val table0 = Table("Table0", setOf(idColumn), HashSet<ForeignKey>())
+
+        val fkColumn = Column("fkId", "INTEGER", 10, primaryKey = false)
+        val table1 = Table("Table1", setOf(idColumn, fkColumn), HashSet<ForeignKey>())
+
+
+        val insertionId = 1001L
+
+        val integerGene = IntegerGene(idColumn.name, 42, 0, 10)
+        val primaryKeyTable0Gene = SqlPrimaryKeyGene(idColumn.name, "Table0", integerGene, 10)
+        val primaryKeyTable1Gene = SqlPrimaryKeyGene(idColumn.name, "Table1", integerGene, 10)
+        val foreignKeyGene = SqlForeignKeyGene(fkColumn.name, insertionId, "Table0", nullable = false)
+
+
+        val insertIntoTable0 = DbAction(table0, setOf(idColumn), insertionId, listOf(primaryKeyTable0Gene))
+        val insertIntoTable1 = DbAction(table1, setOf(idColumn, fkColumn), insertionId, listOf(primaryKeyTable1Gene, foreignKeyGene))
+
+        val (format, baseUrlOfSut, ei) = buildEvaluatedIndividual(mutableListOf(insertIntoTable0, insertIntoTable1))
+
+        val test = TestCase(test = ei, name = "test")
+
+        val writer = TestCaseWriter()
+
+        val lines = writer.convertToCompilableTestCode(format, test, baseUrlOfSut)
+
+        val expectedLines = Lines()
+        expectedLines.add("@Test")
```

Comment:
this can be simplified with ".apply{}", so you do not have to repeat expectedLines